### ISSUE TITLE
<agencyCode>

### ISSUE THREAD

**Issue by SJagodzinski** [SAA-SDT/eac-cpf-schema#87]
## Agency Code
- change element content to text and remove regex pattern
- add attributes
`@audience `
 `@languageOfElement`
 `@scriptOfElement`
`@status`
`@valueURI` 
`@vocabularySource` 
`@vocabularySourceURI` 
`@target`
- review **attributes**:
- Use `@vocabularySource` to encode the standard that the agency code complies with.
- Recommend the use of data value standards and use ISO 15511 as an example in the **Tag Library** and **Best Practise Guide** with text about "good reasons" for use of a data value standard.
- create Schematron validation to check according ISO 15511 regex pattern

## Creator of issue
1. Silke Jagodzinski
2. TS-EAS: EAC-CPF subgroup
3. silkejagodzinski@gmail.com

## Related issues / documents
[`<agencyCode>`: change or remove regex pattern #14](https://github.com/SAA-SDT/eac-cpf-schema/issues/14)
[`<agencyCode>`: add attribute @type #15](https://github.com/SAA-SDT/eac-cpf-schema/issues/15)
[Topic_identifiers.pdf](Topic_identifiers.pdf)
[Identifiers in EAC-CPF](https://docs.google.com/document/d/1RVQ7q1mtWKTpdX0v0PgFyPFNFaGMnumegHvrbwktUUY/edit#heading=h.gy2f20ngdxdt)
[Repository codes and other identifiers](https://drive.google.com/open?id=1GNLY3NgfLZdTWA0PCFGmsajmdL8JOeOZBJVTjZBIZos)

## EAD3 Reconciliation
**Additional EAD 3 attributes**
`@altrender` - Optional
`@audience` - Optional (values limited to: external, internal)
`@encodinganalog` - Optional
`@lang `- Optional
`@localtype` - Optional
`@script` - Optional

**May contain**: [text]

## Context
The code that represents the institution or service responsible for the creation, maintenance and/or dissemination of the EAC-CPF instance.

**May contain**: [text]
**May occur within**: `<maintenanceAgency>`
**Attributes**: `@xml:id` - Optional
**Availability**: Optional, Non-repeatable

## Solution documentation
Rephrasing _Summary_, _Description and Usage_ and _Attribute usage_ needed?

**May contain**: [text]
**May occur within**: `<maintenanceAgency>`
**Attributes**:
`@audience` - optional (values limited to: external, internal)
`@id` - optional
`@languageOfElement` - optional
`@scriptOfElement` - optional
`@status` - optional (values limited to: authorized, alternative)
`@valueURI` - optional
`@vocabularySource` - optional
`@vocabularySourceURI` - optional
`@target` - optional
**Availability**: Optional, not repeatable

- New examples needed
- Topic for Best Practise Guide

## Example encoding
```
<control>
 <maintenanceAgency>
   <agencyCode audience="external" id="agencycode1" languageOfElement="eng" scriptOfElement="lat" status="authorized" target="agencyName1" valueURI="agencyCodeURI" vocabularySource="ISIL" vocabularySourceURI="ISILURI">agency code</agencyCode>
  <agencyName>agency name</agencyName>
 </maintenanceAgency>
</control>
```

**Comment by SJagodzinski** [comment]
EAC-CPF Berlin meeting, 11 March 2020:

Hand a decision over to the Schema team, if @localType can be used in <control> elements.

Hand a discussion on reintroducing @type over to the Schema team.

**Comment by SJagodzinski** [comment]
EAC-CPF Berlin meeting, 12 March 2020:

Suggested that @valueURI, @vocabularysource, and @vocabularysourceURI be added, as optional attributes, to encode the source of the authority

For example:
`<agencycode vocabularysource="VIAF" vocabularysourceURI="[VIAF URI]" valueURI="[URI of original instance]">`

**Comment by SJagodzinski** [comment]
EAC-CPF meeting, 6 November 2020:

Use `@vocabularySource` to encode the standard that the agency code complies with.

**Comment by ailie-s** [comment]
Tag Library Text:

**Summary:** A child element of `<maintenanceAgency>` that provides a code for the institution or service responsible for the creation, maintenance, and/or dissemination of the EAC-CPF instance.
**May contain:** [text]
**May occur within:** **<maintenanceAgency>**
**Attributes:**
`@audience` - optional (values limited to: external, internal)
`@id` - optional
`@languageOfElement` - optional
`@scriptOfElement `- optional
`@status` - optional (values limited to: authorized, alternative)
`@valueURI` - optional
`@vocabularySource` - optional
`@vocabularySourceURI `- optional
**Description and Usage:**
Use `<agencyCode>` to record a code indicating the institution or service responsible for the creation, maintenance and/or dissemination of the EAD instance. Use of `<agencyCode>` is recommended, as the combination of `<agencyCode>` and the required `<recordId>` provide a globally unique identifier for the instance.
It is recommended that the code follow the format of the International Standard Identifier for Libraries and Related Organizations (ISIL: ISO 15511): a prefix, a dash, and an identifier. The code is alphanumeric (characters A-Z, 0-9, solidus, hyphen-minus, and colon) with a maximum of 16 characters. If appropriate to local or national convention, insert a valid ISIL for an institution, whether provided by a national authority (usually the national library) or a service (such as OCLC). If this is not the case then local institution codes may be given with the ISO 3166-1 alpha-2 country code as the prefix to ensure international uniqueness in `<agencyCode>`.
**Attribute Usage:**
Use `@status` with the values "authorized" or "alternative" to declare whether the `<agencyCode>` is using an authorized or alternative value
**See Also:**
Use `<agencyName>` to record the name of the agency.
Use `<otherAgencyCode>` to record any alternative codes representing the agency.
Use `<recordId>` in combination with `<agencyCode>` to provide a globally unique identifier for the EAC-CPF instance.
**Availability:** Optional, not repeatable

**Comment by kerstarno** [comment]
Tested as part of the Schema Team's schema testing:

- `<agencyCode>` as sub-element of `<maintenanceAgency>` is optional
- `<agencyCode>` as sub-element of `<maintenanceAgency>` cannot be repeated
- `<agencyCode>` does not have any sub-elements
- `<agencyCode>` can contain text, which can be any text, i.e. does not have to follow the structure of ISO 15511 and doesn't require an ISO 3166-1 alpha-2 country code as a prefix
- `<agencyCode>` allows for the optional attributes `@audience`, `@id`, `@target`, `@languageOfElement`, `@scriptOfElement`, `@status` (with the values "authorized" or "alternative"), `@valueURI`, `@vocabularySource`, and `@vocabularySourceURI`
- `<agencyCode>` also allows for optional attributes from other namespaces (checked with XLink)

This applies to RNG and XSD schemas. 

The element is ready.

@ailie-s - the TL draft currently is missing `@target`. Furthermore, `@languageOfElement` was listed twice in the comment above. I've edited this here, but you probably would like to check your local TL file.
